apps/judicial-system/digital-mailbox-api/src/app/app.service.ts

```diff
@@ -21,8 +24,54 @@ export class AppService {
     return 'OK'
   }
 
-  async testConnection(): Promise<string> {
+  async testConnection(token: string): Promise<string> {
     //TODO: Audit
+    const verifiedUserToken = await this.verifyIdsToken(token)
+
+    if (!verifiedUserToken.sub) {
+      throw new UnauthorizedException('Invalid token')
+    }
+
     return this.test()
   }
+
+  async verifyIdsToken(token: string) {
```

Comment:
I think we should move this to a guard so it's easier to use for all the endpoints

Reply:
We could start with something along the lines of this 
https://github.com/island-is/island.is/commit/bfb45f65bd6f01f957cc222108066b8f2f6e9c79

I don't know whether we want to have this in our lib or in this project, but could be a start to just have it here

Reply:
I also realized we can use the passport library to handle the verification, a draft of the idea :
https://github.com/island-is/island.is/pull/14882

Should we start with something like this? It also makes it easier to extract the national id to verify that the user can access cases only connected to their national id